bson/src/main/org/bson/Vector.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson;
+
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import static org.bson.assertions.Assertions.assertNotNull;
+import static org.bson.assertions.Assertions.isTrue;
+import static org.bson.assertions.Assertions.isTrueArgument;
+import static org.bson.assertions.Assertions.notNull;
+
+/**
+ * Represents a vector that is stored and retrieved using the BSON Binary Subtype 9 format.
+ * This class supports multiple vector {@link Dtype}'s and provides static methods to create
+ * vectors and methods to retrieve their underlying data.
+ * <p>
+ * Vectors are densely packed arrays of numbers, all the same type, which are stored efficiently
+ * in BSON using a binary format.
+ *
+ * @mongodb.server.release 6.0
+ * @see BsonBinary
+ * @since BINARY_VECTOR
+ */
+public final class Vector {
```

Comment:
Consider adding a codec for Vector, so that they can be included directly in a POJO or record without having to manually wraps it in `BsonBinary`.

Reply:
Thanks for the suggestion! I've added codecs for each vector subclass, along with the corresponding tests.